### ISSUE TITLE
[Doc] Add robots.txt to the pulsar site

### ISSUE THREAD

**Issue by dave2wave** [apache/pulsar#19107]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### What issue do you find in Pulsar docs?

We are missing a robots.txt file in the website

### What is your suggestion?

In order to improve search engine crawls of https://pulsar.apache.org we should include a robots.txt file in the pulsar-site with the static folder.

See https://developers.google.com/search/docs/crawling-indexing/robots/intro

This should refer directly to our existing https://pulsar.apache.org/sitemap.xml

Also we may want to consider changes to the sitemap.

The goal is to have the search engines find recent content over old version docs.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by nodece** [comment]
This issue should move to https://github.com/apache/pulsar-site.

**Comment by tisonkun** [comment]
@nodece No. The site repo doesn't have an issue tab now. We use the main repo to host all the issues.

There's a previous discussion but I forget its link :/
